Gridy/View/GridOverlay.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  GridOverlay.swift
+//  Gridy
+//
+//  Created by Alexandra Ivanova on 29/10/2020.
+//  Copyright © 2020 Alex Ivanova. All rights reserved.
+//
+
+import UIKit
+
+class GridOverlay: UIView {
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.setup()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        self.setup()
+    }
+
+    private func setup() {
+        let maskLayer = CAShapeLayer()
+        maskLayer.frame = self.bounds
+        let radius: CGFloat = self.bounds.width/3
+        let path = UIBezierPath(rect: self.bounds)
+        path.addArc(
+            withCenter: self.center,
+            radius: radius,
+            startAngle: 0.0,
+            endAngle: CGFloat.pi*2,
+            clockwise: true
+        )
+        maskLayer.path = path.cgPath
+        maskLayer.fillRule = CAShapeLayerFillRule.evenOdd
+
+        self.layer.mask = maskLayer
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        self.setup()
```

Comment:
You generally want to setup once. Here you are going to create delete and create a new CAShapeLayer every time `layoutSubviews` is called which is not ideal for the performance of the app. The fix would to keep the same object but update it.

Reply:
I'm not sure how to update the object. I have the same issue with the horizontal/vertical screen constraints. I can setup them but I don't know how to update them every time the screen orientation changes. Any links to point me in right direction?

Reply:
You are already "updating" your object, just not in the right place... Maybe these comments will help you to realize it:
```
private func setup() {
        let maskLayer = CAShapeLayer() // Object creation, not update needed
        maskLayer.frame = self.bounds // Update needed when the frame of the view is updated
        let radius: CGFloat = self.bounds.width/3 // Update needed when the frame of the view is updated
        let path = UIBezierPath(rect: self.bounds) // Update needed when the frame of the view is updated
        path.addArc(
            withCenter: self.center,
            radius: radius,
            startAngle: 0.0,
            endAngle: CGFloat.pi*2,
            clockwise: true
        ) // Update needed when the frame of the view is updated
        maskLayer.path = path.cgPath // Update needed when the frame of the view is updated
        maskLayer.fillRule = CAShapeLayerFillRule.evenOdd // This is a one time setup

        self.layer.mask = maskLayer // This is a one time setup
}
```